pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -448,6 +458,12 @@ func (c *Controller) onNodeEvent(obj interface{}, event model.Event) error {
 		updatedNeeded = true
 		c.Lock()
 		delete(c.nodeInfoMap, node.Name)
+
+		// We should remove the node for all the services that
+		// had a workload there as we don't want to route traffic
+		for h := range c.serviceToWorkloadNodeMap {
+			delete(c.serviceToWorkloadNodeMap[h], node.Name)
```

Comment:
in xcp we dont handle `ConfigUpdate()`(L#493) , we should do that change in edge to syncup node related changes

Reply:
May be we should send `reason` as well at L493 as may be `NodeEvent` or something like that so that in edge we can ignore other `ConfigUpdate` than the nodeEvent one

Reply:
Added `NodeTrigger` reason. Yes. I think it is needed to reduce the updates we process in XCP.

Reply:
Can you please also open an inssue in xcp for handling `ConfigUpdate` callbacks for node realted changes?